kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/FilterHandler.java

```diff
@@ -495,6 +503,40 @@ public String getVirtualClusterName() {
                             Objects.requireNonNull(localTlsCertificate(clientFacingSslHandler)), getPeerTlsCertificate(clientFacingSslHandler)));
         }
 
+        @Override
+        public void clientSaslAuthenticationSuccess(@NonNull String mechanism,
+                                                    @NonNull String authorizedId) {
+            LOGGER.atInfo().setMessage("{}: Filter '{}' announces client has passed SASL authentication using mechanism '{}' and authorizationId '{}'.")
+                    .addArgument(channelDescriptor())
+                    .addArgument(filterDescriptor())
+                    .addArgument(mechanism)
+                    .addArgument(authorizedId)
+                    .log();
+            // dispatch principal injection
+            csc.clientSaslAuthenticationSuccess(mechanism, authorizedId, null);
+        }
+
+        @Override
+        public void clientSaslAuthenticationFailure(@Nullable String mechanism,
+                                                    @Nullable String authorizedId,
+                                                    @NonNull Exception exception) {
+            LOGGER.atInfo()
+                    .setMessage("{}: Filter '{}' announces client has failed SASL authentication using mechanism '{}' and authorizationId '{}'. Cause message {}")
+                    .addArgument(channelDescriptor())
+                    .addArgument(filterDescriptor())
+                    .addArgument(mechanism)
+                    .addArgument(authorizedId)
+                    .addArgument(exception.toString())
```

Comment:
Maybe we should conditionally include the full exception for stacktrace logging at DEBUG evel. Pattern from elsewhere:

```
LOGGER.atWarn()
358-                    .setCause(LOGGER.isDebugEnabled() ? cause : null)
359-                    .log("An exceptionCaught() event was caught by the error handler {}: {}. Increase log level to DEBUG for stacktrace",
360-                            cause.getClass().getSimpleName(), cause.getMessage());
```